Modules/Core/Common/test/itkMathRoundTest.cxx

```diff
@@ -20,7 +20,7 @@
 #include "itkIndex.h"
 
 bool
-math_test_helper(std::string str, bool test)
+math_test_helper(const std::string & str, bool test)
```

Comment:
In this particular case, I would expect any performance gain from using `const std::string &`, because `math_test_helper` is only called with a "literal" as argument. So if it's just about performance, I would use either `const char *` or (preferably) `std::string_view` as parameter type.

I'm interested to know if cppcheck allows passing `std::string_view` _by value_, because it's generally suggested to pass `std::string_view` by value.

Reply:
cppcheck categorizes this under "performance", but this is just test code, so performance isn't an issue anyway.

Reply:
OK, so this proposed change is only meant to make cppcheck happy, right?

Reply:
- This is cool: https://github.com/danmar/cppcheck/pull/3817

So yes, cppcheck _allows_ passing `std::string_view` by value  😃

Reply:
> OK, so this proposed change is only meant to make cppcheck happy, right?

Yes. :)

Reply:
I didn't use `ITK_CPPCHECK_TEST`.  I used `CMAKE_EXPORT_COMPILE_COMMANDS=ON` then invoked with:

`cppcheck --project=compile_commands.json --platform=unspecified --enable=style -q --library=qt --library=posix --library=gnu --library=bsd --library=windows --check-level=exhaustive --suppressions-list=../ITK/ITKcppcheckSuppressions.txt --template='{id},{file}:{line},{severity},{message}' -j22`

Reply:
>I just checked: replacing std::string with std::string_view does indeed make cppcheck happy, in this case :)

OK, changed.